test/widgets.js

```diff
@@ -0,0 +1,127 @@
+        aposLocale: 'en:published',
+        aposDocId: 'goodPageId',
+        type: 'args-good-page',
+        slug: '/good-page',
+        visibility: 'public',
+        path: `${homeId.replace(':en:published', '')}/good-page`,
+        level: 1,
+        rank: 0,
+        main: {
+          _id: 'randomAreaId1',
+          items: [
+            {
+              _id: 'randomWidgetId1',
+              snippet: 'You can control what happens when the text reaches the edges of its content area using its attributes.',
+              metaType: 'widget',
+              type: 'args'
+            }
+          ],
+          metaType: 'area'
+        }
+      },
+      {
+        _id: 'badPageId:en:published',
+        aposLocale: 'en:published',
+        aposDocId: 'badPageId',
+        type: 'args-bad-page',
+        slug: '/bad-page',
+        visibility: 'public',
+        path: `${homeId.replace(':en:published', '')}/bad-page`,
+        level: 1,
+        rank: 0,
+        main: {
+          _id: 'randomAreaId2',
+          items: [
+            {
+              _id: 'randomWidgetId2',
+              snippet: 'You can control what happens when the text reaches the edges of its content area using its attributes.',
+              metaType: 'widget',
+              type: 'args'
+            }
+          ],
+          metaType: 'area'
+        }
+      }
+    ];
+
+    // Insert draft versions too to match the A3 data model
+    const draftItems = await apos.doc.db.insertMany(testItems.map(item => ({
+      ...item,
+      aposLocale: item.aposLocale.replace(':published', ':draft'),
+      _id: item._id.replace(':published', ':draft')
+    })));
+    assert(draftItems.result.ok === 1);
+    assert(draftItems.insertedCount === 2);
+
+    const items = await apos.doc.db.insertMany(testItems);
+
+    assert(items.result.ok === 1);
+    assert(items.insertedCount === 2);
+  });
+
+  it('should be able to render page template with well constructed area tag', async function() {
+    const req = apos.task.getAnonReq();
+
+    const goodPage = await apos.page.find(req, { slug: '/good-page' }).toObject();
+    goodPage.metaType = 'doc';
+
+    const result = await apos.modules['args-good-page'].renderPage(req, 'page', { page: goodPage });
+
+    assert(result.indexOf('<h2>Good args page</h2>') !== -1);
+    assert(result.indexOf('<p>You can control what happens when the text reaches the edges of its content area using its attributes.</p>') !== -1);
+    assert(result.indexOf('<li>color: 🟣</li>') !== -1);
+  });
+
+  it('should error while trying to render page template with poorly constructed area tag', async function() {
+    const req = apos.task.getAnonReq();
+
+    const goodPage = await apos.page.find(req, { slug: '/bad-page' }).toObject();
+    goodPage.metaType = 'doc';
+
+    try {
+      await apos.modules['args-bad-page'].renderPage(req, 'page', { page: goodPage });
+      assert(false);
+    } catch (error) {
+      assert(error);
```

Comment:
Sounds like it's doing the job, but I don't actually see why the parser fails on this, so it may be worth inspecting the error to make sure it's saying something useful.

Reply:
The error is caused in the Bad Page template, passing three arguments to the area tag before `with`. It'd definitely be better if everything was in the test file, but it seems we're already using fake modules and their templates in other tests already.

Reply:
What I mean is that I'm not clear on where you are *checking* for too many arguments, and so it's not clear to me if the actual error message is a good error message.

Reply:
I see. I might need a hand with this. I thought the `await apos.modules['args-bad-page']` line would throw an error, but it doesn't (there, at least). The response is simply the error page template.

Is there a way we can capture the template render error, "Too many arguments were passed to the 'baddies' area before the 'with' keyword."?